### ISSUE TITLE
The shell prompt runs off the bottom end of the screen on iPhone 12 Mini

### ISSUE THREAD

**Issue by m4rkw** [ish-app/ish#1059]
When using an external keyboard the prompt runs off the end of the screen:

https://gyazo.com/522bdd823f5d065422ac416876b494fc

Not sure if this is device-specific but it occurs on an iPhone 12 Mini.

**Comment by m4rkw** [comment]
I found a workaround, put this in ~/.bash_profile:

`export PS1="$PS1                                 \[\033[1A\]\[\033[5C\]'`

Uses spaces to run the prompt to the next line and then uses escape sequences to return it to where it was. This works for portrait mode using Menlo font size 16. Needs adjusting for horizontal or different font sizes but it's easy enough to change. You need enough spaces to push the prompt onto the next line, then the 1A escape moves it to the start of the previous line, then 5C moves it 5 characters right to be where it should be.

**Comment by 62f** [comment]
Gives syntax error:
*
export PS1="$PS1 \[\033[1A\]\[\033[5C\]'
*

**Comment by m4rkw** [comment]
Quotes are mismatched.

**Comment by 62f** [comment]
> Quotes are mismatched.

Yes I know. In landscape view, prompt also sometimes runs off the bottom of Alpine 3.10 -ash shell on my iPhone 5s; when it’s not busy duplicating lines. Yes, I did just close that other issue because now it’s partly duplicating this bug.

**Comment by t57root** [comment]
This issue also occurred on an iPad Pro.

**Comment by tbodt** [comment]
On iPhone with rounded corners (but not iPhone without rounded corners, or iPad regardless of rounded corners) when a hardware keyboard is present and a UIInputView is present, the keyboard is reported as not being there at all instead of taking up the space of the UIInputView. god this is so buggy

**Comment by laseryuan** [comment]
I get around it by nesting the terminal into a tmux session. Set the height of tmux status bar large enough:
`set-option -g status 2`

**Comment by 62f** [comment]
Seems like "nesting...session" may require having 2 devices.
